src/Indice.Features.Identity.Core/Data/Stores/ExtendedUserStore.cs

```diff
@@ -223,4 +223,19 @@ await Context.Set<UserPassword>()
         }
         return IdentityResult.Success;
     }
+
+    /// <inheritdoc/>
+    public async Task SetUsernameChangeAsync(TUser user, string previousUsername, DateTimeOffset dateCreated, CancellationToken cancellationToken = default) {
+        cancellationToken.ThrowIfCancellationRequested();
+        ThrowIfDisposed();
+
+        await Context.Set<UserUsername>()
+            .AddAsync(new UserUsername {
```

Comment:
The method `AddAsync` does not go with save changes because it sends the `INSERT` statement on the spot. It is usually preferred only when there are Database calculated properties that we cannot otherwise get.